src/app/notifications/types.ts

```diff
@@ -25,6 +25,13 @@ export interface ToastNotification {
   timestamp: number;
   destination: 'studyGuides' | 'myHighlights' | 'page';
   shouldAutoDismiss: boolean;
+  errorId?: string;
+}
+
+export interface ToastMeta {
+  destination: ToastNotification['destination'];
+  shouldAutoDismiss?: boolean;
+  errorId?: string;
```

Comment:
What about making `ToastNoftification extends ToastMeta` and then remove redundant fields?